impl/httpMetricsManager.go

```diff
@@ -30,10 +68,36 @@ func (m *HttpMetricsManager) GetThroughput() int {
 	return m.requestSent
 }
 
+// return the sampletime(duration) at input percentile
+func (m *HttpMetricsManager) GetDurationAt(thre int) time.Duration {
+	if !sort.IsSorted(m.records) {
+		sort.Sort(m.records)
+	}
+	queryIndx := len(m.records) * thre / 100
+	return m.records[queryIndx].duration / time.Millisecond
+}
+func (m *HttpMetricsManager) GetMeanDuration() int {
+	sum := m.durationSum.sum
+	for i := m.durationSum.index + 1; i < len(m.records); i++ {
+		sum += m.records[i].duration / time.Millisecond
+	}
+	m.durationSum.index = len(m.records)
+	m.durationSum.sum = sum
+	return int(sum) / len(m.records)
+}
+
+// Data for each response shoud be periodically dump to disk
+func (m *HttpMetricsManager) DumpToDisk() {
```

Comment:
The concern I have here is on the number of `record` in long run. To get the other latency/sample time metrics, each records is needed to store in memory for calculation. Say a struct is 16 bytes (one int64 duration + other vars) and we run a job at 10request/sec for 10 hours. Then the memory taken by this is quite huge: 16*10*(60*60*10) /1024 = 5625 Mb. So maybe dumping some to disk is a unavoidable choice. Or we can use binning as I suggest above.